diff --git a/Platform/ApfsDriverLoader/ApfsDriverLoader.c b/Platform/ApfsDriverLoader/ApfsDriverLoader.c
index 42b98d5..f130f98 100755
--- a/Platform/ApfsDriverLoader/ApfsDriverLoader.c
+++ b/Platform/ApfsDriverLoader/ApfsDriverLoader.c
@@ -208,7 +208,7 @@ ReadDisk (
   IN UINT32                 MediaId,
   IN UINT64                 Offset,
   IN UINTN                  BufferSize,
-  OUT VOID                  *Buffer
+  OUT UINT8                 *Buffer
   )
 {
   EFI_STATUS  Status;
@@ -343,10 +343,10 @@ ApfsDriverLoaderStart (
   UINT32                      PartitionNumber              = 0;
   UINT32                      PartitionEntrySize           = 0;
   EFI_BLOCK_IO_MEDIA          *Media                       = NULL;
-  UINT32                      Lba                          = 0;
-  VOID                        *Block                       = NULL;
+  UINTN                       Lba                          = 0;
+  UINT8                       *Block                       = NULL;
   EFI_PARTITION_ENTRY         *ApfsGptEntry                = NULL;
-  VOID                        *ApfsBlock                   = NULL;
+  UINT8                       *ApfsBlock                   = NULL;
   UINT64                      JsdrOffset                   = 0;
   UINT64                      JsdrPtr                      = 0;
   APFS_EFI_BOOT_RECORD        *JsdrBlock                   = NULL;
